### PR TITLE
[sqllogictest] Create empty tables

### LINKED ISSUE #4396: Support CREATE TABLE table_name(...schema_fields)
**Describe the solution you'd like**
Support the `CREATE TABLE` expression, e.g.
`CREATE TABLE t1(a INTEGER, b INTEGER, c INTEGER, d INTEGER, e INTEGER);`

Executing this with `datafusion-cli` shows:
```bash
DataFusion CLI v14.0.0
❯ CREATE TABLE t1(a INTEGER, b INTEGER, c INTEGER, d INTEGER, e INTEGER);
NotImplemented("Only `CREATE TABLE table_name AS SELECT ...` statement is supported")
```

**Additional context**
The ability to create a table with one statement and then insert values into it throughout a sessions forms the basis of a lot of [sqllogictests](https://www.sqlite.org/sqllogictest/file?name=test/select1.test&ci=tip) we want to incorporate into testing. See https://github.com/apache/arrow-datafusion/issues/4248

### PR DESCRIPTION
**Pull request by melgenek**
# Which issue does this PR close?

Closes #4396.

# Rationale for this change

# What changes are included in this PR?

The change allows the creation of empty tables in sqllogictest.

# Are these changes tested?

Github Actions run sqllogictest.

# Are there any user-facing changes?

Planner's `build_schema` was made public. I am not sure if this is a problem.

@xudong963 @mvanschellebeeck

### CODE REVIEW

File: datafusion/core/tests/sqllogictests/test_files/ddl.slt [hunk 1]
```diff
@@ -419,3 +419,49 @@ SELECT * FROM aggregate_simple order by c1 DESC LIMIT 1;
...
+CREATE TABLE table_without_values(field1 BIGINT, field2 BIGINT);
+
+
+# 'IF NOT EXISTS' cannot coexist with 'REPLACE'
+statement error Execution error: 'IF NOT EXISTS' cannot coexist with 'REPLACE'
+CREATE OR REPLACE TABLE IF NOT EXISTS table_without_values(field1 BIGINT, field2 BIGINT);
+
+# Should insert into an empty table
+statement ok
+insert into table_without_values values (1, 2), (2, 3), (2, 4);
```
Comment (alamb) [inline datafusion/core/tests/sqllogictests/test_files/ddl.slt:445]: 👍

File: datafusion/sql/src/planner.rs [hunk 2]
```diff
@@ -114,7 +114,7 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
         }
     }
 
-    pub(crate) fn build_schema(&self, columns: Vec<SQLColumnDef>) -> Result<Schema> {
+    pub fn build_schema(&self, columns: Vec<SQLColumnDef>) -> Result<Schema> {
```
Comment (alamb) [inline datafusion/sql/src/planner.rs:117]: I think this is fine to make public 👍

**Review by alamb — APPROVED** [review]
Looks good to me -- thank you @melgenek

**Comment by alamb** [comment]
I really appreciate your efforts to drive sqllogictests forward @melgenek  -- it is great working with you

**Comment by xudong963** [comment]
Cool, thanks @melgenek

**Pull request merged**